CatalogService.Core/Handlers/Categories/GetCategoriesByProductHandler.cs

```diff
@@ -0,0 +1,24 @@
+﻿using CatalogService.Core.Entities;
+using CatalogService.Infrastructure.Dto;
+using CatalogService.Infrastructure.MsSql.Categories;
+
+namespace CatalogService.Core.Handlers.Categories;
+
+public interface IGetCategoriesByProductHandler
+{
+    Task<IEnumerable<Category>> HandleAsync(int productId);
+}
+
+public class GetCategoriesByProductHandler : IGetCategoriesByProductHandler
+{
+    private readonly ICategoryRepository _categoryRepository;
+
+    public GetCategoriesByProductHandler(ICategoryRepository categoryRepository) => 
+        _categoryRepository = categoryRepository;
+
+    public async Task<IEnumerable<Category>> HandleAsync(int productId)
```

Comment:
The common rule is to return the most concreate type and accept the most general one, what is the benefit to return an iterator?

Reply:
this is a very good point, thanks, fixed